### ISSUE TITLE
Multi-threaded generation of XYZ Vector Tiles

### ISSUE THREAD

**Issue by gio0v** [qgis/QGIS#53875]
### Feature description

When I use the most recent version of QGIS to generate XYZ Raster Tiles, it automatically employs all the the CPU cores and threads available. However, the generation of XYZ Vector Tiles runs in a single CPU thread - which represents such a severe performance loss that it makes Vector Tile generation with QGIS an impossibility when working with large area layers.

**Comment by nirvn** [comment]
@alexbruy , has that been fixed by your work porting algorithms to c++?

**Comment by alexbruy** [comment]
@nirvn no. Vector tiles generation algorithm was implemented in C++ from the beginning.

**Comment by nirvn** [comment]
Oh doh, in my haste I skipped the crucial _vector_  keyword :)

**Comment by p-sims** [comment]
I'd just like echo request for multi thread rendering of vector layers. It seems that in QGIS 3.34 on Linux (Pop! OS 22) that all vector rendering and many file/layer/project load/save tasks are all single thread, so for project with big vector layers to load QGIS doesn't utilise most of the computer hardware for many operations. 

Also GNOME often brings up the "This application is not responding" pop up during project load, layer save as operation, which are maxing out a single thread for long periods and making UI unresponsive in the mean time. Prehaps that's a seperate issues but I guess such operation happen on UI thread rather than background worker.
